projects/novo-elements/src/elements/data-table/cell-headers/data-table-checkbox-header-cell.component.ts

```diff
@@ -70,6 +74,19 @@ export class NovoDataTableCheckboxHeaderCell<T> extends CdkHeaderCell implements
   }
 
   public onClick(): void {
-    this.dataTable.selectRows(!this.checked);
+    if (this.isAtLimit()) {
+      this.toaster.alert({
+        theme: 'danger',
+        position: 'fixedTop',
+        message: 'Error, more than 500 files are not able to be selected at one time',
```

Comment:
this error looks specific to files but since this data-table is not, it would be good to have a generic message here